Automattic/wp-calypso
# Jetpack Manage: Adding dataviews filter and search functionality

Related to https://github.com/Automattic/jetpack-manage/issues/287

## Proposed Changes

* This PR adds filter and search functionality to the sites dashboard views.
* Clicking on reset will remove URL params (relevant for both search and filters) and show all sites.
* The filter does not include favorites, in line with the filter capabilities on the existing dashboard page and given that the favorites filter currently leads to a new page as well. We will check in with designers about whether to implement this later on.

Future to-dos:

* Possibly for a future PR - remove the 'is' from the filter dropdown (eg. 'Status Backup Failed' instead of 'Status is Backup Failed').

## Testing Instructions

* To test, apply this PR and navigate to `http://jetpack.cloud.localhost:3000/sites`. From here, click on the filter dropdown and select any option. The relevant filtered options should then show on the page, and the relevant filter will show as the selected filter.
* Click on reset and you should be sent back to the sites page.
* To test search, search for any characters within a site name (currently the only search option), and the search results should refresh immediately. Reset will remove the search query.

Screenshot of filter unselected:
<img width="510" alt="Screenshot 2024-02-21 at 16 08 42" src="https://github.com/Automattic/wp-calypso/assets/16754605/26514175-3879-4b8c-bb34-91284a508f4e">

Screenshot of filter selected:
<img width="570" alt="Screenshot 2024-02-21 at 16 08 54" src="https://github.com/Automattic/wp-calypso/assets/16754605/d1703cbf-8a79-4990-8495-2c2dd04814c8">



## Pre-merge Checklist
